arangod/MMFiles/MMFilesCollection.cpp

```diff
@@ -2462,27 +2460,19 @@ bool MMFilesCollection::dropIndex(TRI_idx_iid_t iid) {
 bool MMFilesCollection::removeIndex(TRI_idx_iid_t iid) {
   WRITE_LOCKER(guard, _indexesLock);
 
-  size_t const n = _indexes.size();
-
-  for (size_t i = 0; i < n; ++i) {
-    auto idx = _indexes[i];
-
+  for (auto& idx : _indexes) {
     if (!idx->canBeDropped()) {
       continue;
     }
-
     if (idx->id() == iid) {
       // found!
       idx->drop();
-
-      _indexes.erase(_indexes.begin() + i);
-
       // update statistics
       MMFilesIndex* midx = static_cast<MMFilesIndex*>(idx.get());
       if (midx->isPersistent()) {
         --_persistentIndexes;
       }
-
+      _indexes.erase(idx);
```

Comment:
again, is this safe here? we are invalidating the iterator `idx`, which we use in the outer for loop.

Reply:
Looks safe to me as far as next line terminates the loop